deps/build.jl

```diff
@@ -76,6 +76,21 @@ if !custom_library
         # Download and install binaries
         install(dl_info...; prefix=prefix, force=true, verbose=verbose)
     end
+
+    # Warn the user that the packeges from BinaryProvider are incompatible with
+    # their installation of Julia. While this is mentioned in README.md users
+    # might still try. In that case, BinaryProvider (--verbose) fails with
+    # a confusing message "could not locate," instead of "could not dlopen."
+    usable(p) = satisfied(p; verbose=verbose, isolate=true)
+    failed = [p.libnames for p in products if !usable(p)]
+    if !isempty(failed)
+        error(
+            "The following dependencies could not be satisfied by binaries " *
+            "from BinaryProvider: \n$failed.\nIn case You are using non-" *
+            "official Julia binaries, try compiling SCS from source.\nMake " *
+            "sure Julia and SCS are linked against the same OpenBLAS library.\n"
```

Comment:
Or suggest that they use the official `jll` ones that are automatically downloaded? We should probably only run `build.jl` if the following is true
https://github.com/jump-dev/SCS.jl/blob/6611207d8759cb4a7a1f8c10bcff606f0ccb471a/src/SCS.jl#L3

So something like the following at the top of `build.jl`:
```Julia
if !(haskey(ENV, "JULIA_SCS_LIBRARY_PATH") || VERSION < v"1.3")
    @info("Installing default SCS binaries")
    exit(0)
end
```

Reply:
Maybe You are right, but I do not understand.

I will give you an example to make sure we are on the same page:
I am running on _ArchLinux_. I have installed _Julia_ as a prebuilt package from the package manager _pacman_ (the official package manager of _ArchLinux_, which provides packages in a "rolling distro" style). Similarly, I have installed _OpenBLAS_ from _pacman_. In this case, running `]add SCS` in the _Julia REPL_ will fail because `libopenblas64_.so: cannot open shared object file: No such file or directory`. As I understand it, this is because the library binary `libscsindir` from _BinaryProvider_ is built and linked against an incompatible version of _OpenBLAS_. On the other hand, building `libscsindir` manually works.

Are you saying that the official `jll` binaries can avoid such issues and that they are downloaded before `build.jl` is executed? If so, then I agree with you. That would be a much better solution.

Reply:
> Are you saying that the official jll binaries of libscs can avoid such issues and that they are downloaded before build.jl is executed?

Yes. The build script should only be run on Julia 1.2 or less, or if you wish to provide a non-default binary via the `JULIA_SCS_LIBRARY_PATH` environment variable.

Arch can be finicky sometimes. Can you try the change I suggested and see if it worked?